library/sanitize.inc.php

```diff
@@ -47,4 +47,32 @@ function basename_international($path){
   return $decoded_file_name;
 }
 
+
+/**
+ * This function detects a MIME type for a file and check if it in the white list of the allowed mime types.
+ * @param string $file - file location.
+ * @param array|null $whiteList - array of mime types that allowed to upload.
+ */
+$white_list = null;
+function isWhiteFile($file, &$white_list){
```

Comment:
In the case of multiple upload the isWhiteList function will run multiple times therefore I store the white list in a variable above function to prevent double requests from database.